### PR TITLE
Check that all pre-commits are synchronized code<>docs

### PR DESCRIPTION
**Pull request by potiuk**
Until pre-commit implements export of all configured
checks, we need to maintain the list manually updated.

We check both - pre-commit list in breeze-complete and
descriptions in STATIC_CODE_CHECKS.rst

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -412,3 +412,9 @@ repos:
         language: system
         files: \.mermaid$
         require_serial: true
+      - id: pre-commit-descriptions
+        name: Check if pre-commits are descrbed
```
Comment (feluelle) [inline .pre-commit-config.yaml]:
```suggestion
        name: Check if pre-commits are described
```

**Review by feluelle — COMMENTED** [review]
LGTM, besides a typo.

File: scripts/ci/pre_commit/pre_commit_check_pre_commits.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    fi
+    if [[ ! ${_BREEZE_ALLOWED_STATIC_CHECKS} == *${pre_commit}* ]]; then
+        error="true"
+        >&2 echo "Pre-commit ${pre_commit} is not present in breeze-complete in _BREEZE_ALLOWED_STATIC_CHECKS"
+    fi
+done
+
+if [[ ${error} == "true" ]]; then
+  >&2 echo
+  >&2 echo "Error: Some pre-commits are not synchronized!"
```
Comment (kaxil) [inline scripts/ci/pre_commit/pre_commit_check_pre_commits.sh]: Can we add more details here, like add/delete entry in breeze.rst and breeze-complete

Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_check_pre_commits.sh]: Good point!

Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_check_pre_commits.sh]:
```
Check if pre-commits are described.................................................................................................Failed
- hook id: pre-commit-descriptions
- exit code: 1

ERROR: Pre-commit airflow-config-yaml is not described in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst

FIX: Please add airflow-config-yaml in the table in the 'Pre-commit hooks' chapter in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst


ERROR: Pre-commit base-operator is not described in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst

FIX: Please add base-operator in the table in the 'Pre-commit hooks' chapter in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst


ERROR: Pre-commit base-operator is missing in _BREEZE_ALLOWED_STATIC_CHECKS variable in breeze-complete

FIX: Please add base-operator in the table in the '_BREEZE_ALLOWED_STATIC_CHECKS' constant in /home/jarek/code/airflow/breeze-complete


ERROR: Pre-commit bat-tests is not described in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst

FIX: Please add bat-tests in the table in the 'Pre-commit hooks' chapter in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst


ERROR: Pre-commit black is not described in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst

FIX: Please add black in the table in the 'Pre-commit hooks' chapter in /home/jarek/code/airflow/STATIC_CODE_CHECKS.rst


ERROR: Pre-commit check-hooks-apply is missing in _BREEZE_ALLOWED_STATIC_CHECKS variable in breeze-complete

FIX: Please add check-hooks-apply in the table in the '_BREEZE_ALLOWED_STATIC_CHECKS' constant in /home/jarek/code/airflow/breeze-complete


Some pre-commits are not synchronized! Please fix the errors above!

```

**Review by turbaszek — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Pull request merged**